src/openrct2/ride/Ride.cpp

```diff
@@ -2163,21 +2163,21 @@ void ride_set_vehicle_colours_to_random_preset(Ride* ride, uint8_t preset_index)
         assert(preset_index < presetList->count);
 
         ride->colour_scheme_type = RIDE_COLOUR_SCHEME_ALL_SAME;
-        vehicle_colour* preset = &presetList->list[preset_index];
-        ride->vehicle_colours[0].Body = preset->main;
-        ride->vehicle_colours[0].Trim = preset->additional_1;
-        ride->vehicle_colours[0].Ternary = preset->additional_2;
+        VehicleColour* preset = &presetList->list[preset_index];
+        ride->vehicle_colours[0].Body = preset->Body;
+        ride->vehicle_colours[0].Trim = preset->Trim;
+        ride->vehicle_colours[0].Ternary = preset->Ternary;
```

Comment:
ride->vehicle_colours[0] = *preset;

Reply:
Thanks! It's been a while since I used C++. Trying to get back into it.